app/db/sql/silo.js

```diff
@@ -0,0 +1,21 @@
+const { connect } = require("../connect");
+
+async function list({ entityType, journeyId }) {
+  const knex = connect(entityType);
+  let query = `
+    SELECT * from core.silo s
+    LEFT JOIN (SELECT "siloTypeId", name as "siloTypeName" from core."siloType") st
+    ON s."siloTypeId" = st."siloTypeId"
+  `;
+  if (journeyId) {
+    query += `
+      WHERE "journeyId" = ${journeyId}
```

Comment:
I wonder if we might be opening up a risk to SQL injection here..? Perhaps something like the following might be more secure?
```js
  if (journeyId) {
    query += `
      WHERE "journeyId" = :journeyId
    `;
  } else {
    journeyId = '';
  }
  const res = await knex.raw(query, { journeyId });
```
Although I'm sure it's possible to do something tidier than what I've done here...

Reply:
Thanks @john-coxon! I copied this from elsewhere so I've gone ahead and fix all the others too. Hopefully this last commit will have covered all unsafe string interpolation.

Reply:
That's great! Thanks Ben, and thanks for clearing up the original :+1: